src/client/Microsoft.Identity.Client/AuthScheme/PoP/PopBindingResolver.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.AuthScheme.PoP;
+using Microsoft.Identity.Client.Internal;
+using Microsoft.Identity.Client.Internal.ClientCredential;
+using Microsoft.Identity.Client.Internal.Requests;
+
+namespace Microsoft.Identity.Client.AuthScheme.PoP
+{
+    /// <summary>
+    /// Central place for validating mTLS Proof‑of‑Possession pre‑conditions
+    /// and wiring up <see cref="MtlsPopAuthenticationOperation"/>.
+    /// </summary>
+    internal static class PopBindingResolver
```

Comment:
nit: not a big fan of static classes. Should this be a method on `AcquireTokenCommonParameters` instead?

Reply:
fixed